website: plannable import docs (#33266)
hashicorp/terraform: website/docs/cli/commands/plan.mdx

  * 1 = Error
  * 2 = Succeeded with non-empty diff (changes present)

> - `-generate-config-out=PATH` - (Experimental) If import blocks are present in configuration, instructs Terraform to generate HCL for any imported resources not already present. The configuration is written to a new file at PATH, which must not already exist. Terraform may still attempt to write configuration if the plan errors.

Reviewer: ```
% terraform version
Terraform v1.5.0-beta2
on darwin_arm64
+ provider registry.terraform.io/kreuzwerker/docker v3.0.2
```

> - `-generate-config-out=PATH` - (Experimental) If import blocks are present in configuration, instructs Terraform to generate HCL for any imported resources not already present. The configuration is written to a new file at PATH, which must not already exist. Terraform may still attempt to write configuration if the plan errors.

Author: I've updated the text - what do you think? This section duplicates the command help text itself, so we should update that when this is correct.

> - `-generate-config-out=PATH` - (Experimental) If import blocks are present in configuration, instructs Terraform to generate HCL for any imported resources not already present. The configuration is written to a new file at PATH, which must not already exist, or Terraform will error. If the plan fails for another reason, Terraform may still attempt to write configuration.

Reviewer 2: general question -- will apply also support this and work seamlessly? we have a note on the apply page that it supports all plan options. i wonder if we need to give any addtl guidance on that

Author: This flag is only supported for the `plan` command. There are two categories of `plan` command flags: Plan Customization Options, which are shared with `apply`, and Other Options, which are not. Another example of an option supported by `plan` but not `apply` is `-detailed-exitcode`.

When generating configuration, we want users to run an explicit `plan` step and review the configuration before `apply`. So for example, it is not valid (or possible) to generate configuration during a `terraform apply -auto-approve` - if you try to do this with unmatched import blocks then it will fail and ask you to add config.

> - `-generate-config-out=PATH` - (Experimental) If `import` blocks are present in config, instructs Terraform to generate HCL for any imported resource that does not already have configuration. The config will be written to a file at PATH, which must not already exist. If the plan exits with an error, Terraform may still attempt to write config. After the plan has completed, review the generated config.

Reviewer 3: [Suggestion to match your other PR](https://github.com/hashicorp/terraform/pull/33279/files):
```suggestion
- `-generate-config-out=PATH` - (Experimental) If import blocks are present in configuration, instructs Terraform to generate HCL for any imported resources not already present. The configuration is written to a new file at PATH, which must not already exist. Terraform may still attempt to write configuration if the plan errors.
```
